### PR TITLE
Created Submission Counter Model

### PR DESCRIPTION
**Pull request by JacquelineMorrissette**
Created a model and celery task to count the submissions of each user for each month to be used in a Django Admin Report

Part of kobotoolbox/kpi#3069

### CODE REVIEW

File: onadata/apps/logger/tests/models/test_submission_counter.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        
+
+    def test_data_retrieval(self):
+        """
+        Test that the data stored is the same as the data expected
+        """
+        create_monthly_counters()
+        counter = SubmissionCounter.objects.get(pk=1)
+        today = datetime.now()
+        print(today.year, today.month)
```
Comment (noliveleger) [inline onadata/apps/logger/tests/models/test_submission_counter.py:46]: 👀  , debug line?

File: onadata/apps/logger/models/instance.py [hunk 2]
```diff
@@ -90,6 +91,28 @@ def update_xform_submission_count(sender, instance, created, **kwargs):
         )
 
 
+def update_user_submissions_counter(sender, instance, created, **kwargs):
+    if not created:
+        return
+    if getattr(instance, 'defer_counting', False):
+        return
```
Comment (noliveleger) [inline onadata/apps/logger/models/instance.py:98]: Not sure if it is needed.

File: onadata/apps/logger/models/submission_counter.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import datetime
```
Comment (noliveleger) [inline onadata/apps/logger/models/submission_counter.py:1]: I think it's not needed in python3 anymore, but I usually add `# coding: utf-8` on the first line to define the encoding.
I would keep it to be consistent with old code.

File: onadata/apps/logger/tasks.py [hunk 4]
```diff
@@ -5,14 +5,23 @@
...
 from io import BytesIO
 
-from celery import shared_task
+from celery import task, shared_task
 from dateutil import relativedelta
 from django.contrib.auth.models import User
 from django.core.files.storage import get_storage_class
 from django.core.management import call_command
 
+from django.contrib.auth.models import User
```
Comment (noliveleger) [inline onadata/apps/logger/tasks.py:14]: This import is already present on line 10. You can safely remove this one

File: onadata/apps/logger/tests/models/test_submission_counter.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+
+from mock import patch
+
+from onadata.apps.main.tests.test_base import TestBase
```
Comment (noliveleger) [inline onadata/apps/logger/tests/models/test_submission_counter.py:5]: alphabetization ;-)

**Review by noliveleger — CHANGES_REQUESTED** [review]

**Review by noliveleger — APPROVED** [review]

File: onadata/apps/logger/models/instance.py (hunk 2 above)
Comment (JacquelineMorrissette) [reply onadata/apps/logger/models/instance.py:98]: I tested it with or without. Submissions are likely to be counted twice without it

File: onadata/apps/logger/models/submission_counter.py (hunk 3 above)
Comment (JacquelineMorrissette) [reply onadata/apps/logger/models/submission_counter.py:2]: Resolving because it looks like you took care of it 😄

**Pull request merged**